projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DataJobsSynchronizer.java

```diff
@@ -59,25 +69,48 @@ public DataJobsSynchronizer(
    */
   public void synchronizeDataJobs() throws ApiException {
     ThreadPoolTaskExecutor taskExecutor = initializeTaskExecutor();
-    Set<String> cronJobs = dataJobsKubernetesService.listCronJobs();
+    Set<String> dataJobDeploymentNamesFromKubernetes = dataJobsKubernetesService.listCronJobs();
+    Iterable<DataJob> dataJobsFromDB = jobsRepository.findAll();
+
+    Map<String, DesiredDataJobDeployment> desiredDataJobDeploymentsFromDBMap =
+        desiredJobDeploymentRepository.findAll().stream()
+            .collect(
+                Collectors.toMap(DesiredDataJobDeployment::getDataJobName, Function.identity()));
+
+    Map<String, ActualDataJobDeployment> actualDataJobDeploymentsFromDBMap =
+        actualJobDeploymentRepository.findAll().stream()
+            .collect(
+                Collectors.toMap(ActualDataJobDeployment::getDataJobName, Function.identity()));
 
-    jobsRepository
-        .findAll(
-            Sort.by(
-                Sort.Direction.ASC,
-                DataJob_.DATA_JOB_DEPLOYMENT + "." + DataJobDeployment_.DEPLOYMENT_VERSION_SHA))
-        .forEach(dataJob -> taskExecutor.execute(() -> synchronizeDataJob(dataJob, cronJobs)));
+    dataJobsFromDB.forEach(
+        dataJob ->
+            taskExecutor.execute(
+                () ->
+                    synchronizeDataJob(
+                        dataJob,
+                        desiredDataJobDeploymentsFromDBMap.get(dataJob.getName()),
+                        actualDataJobDeploymentsFromDBMap.get(dataJob.getName()),
+                        dataJobDeploymentNamesFromKubernetes.contains(dataJob.getName()))));
 
     taskExecutor.shutdown();
   }
 
   // Default for testing purposes
-  void synchronizeDataJob(DataJob dataJob, Set<String> cronJobs) {
-    if (dataJob.getDataJobDeployment() != null) {
-      // Sends notification only when the deployment is initiated by the user.
+  void synchronizeDataJob(
+      DataJob dataJob,
+      DesiredDataJobDeployment desiredDataJobDeployment,
+      ActualDataJobDeployment actualDataJobDeployment,
+      boolean isDeploymentPresentInKubernetes) {
+    if (desiredDataJobDeployment != null) {
       boolean sendNotification =
-          StringUtils.isEmpty(dataJob.getDataJobDeployment().getDeploymentVersionSha());
-      deploymentService.updateDeployment(dataJob, sendNotification, cronJobs);
+          true; // TODO [miroslavi] sends notification only when the deployment is initiated by the
+                // user.
+      deploymentService.updateDeployment(
```

Comment:
Is this going to check if the current cron job is the same as the desired state or update it regardless? If the current cron job and the desired state are the same, we don't have to update it....

Reply:
Yes, it is going to update the deployment only if the current cron job and desired state are different - https://github.com/vmware/versatile-data-kit/pull/2714/files#diff-04cc784f09ff0331c51fd011b030b81cb027e8f1531cbfedffa8e7e5df47eca2R202